frontend/src/libs/packages/map/map.package.ts

```diff
@@ -9,36 +9,76 @@ import {
 } from './libs/constants/constants.js';
 import { rotateImg } from './libs/helpers/rotate-img.helper.js';
 import { type IMapService } from './libs/interfaces/interfaces.js';
+import { type MapServiceParameters } from './libs/types/types.js';
 import mapStyle from './map.config.json';
 
-type Constructor = {
-  mapElement: HTMLDivElement;
-  center?: google.maps.LatLngLiteral;
-  zoom: number;
+type Constructor = MapServiceParameters & {
+  extraLibraries?: {
+    geocoding: google.maps.Geocoder;
+    routes: google.maps.DistanceMatrixService;
+    directionsService: google.maps.DirectionsService;
+    directionsRenderer: google.maps.DirectionsRenderer;
+  };
 };
+let map: google.maps.Map | null = null;
+
+const markers: google.maps.Marker[] = [];
 
 class MapService implements IMapService {
   private map: google.maps.Map | null = null;
 
-  private directionsService: google.maps.DirectionsService;
+  private directionsService!: google.maps.DirectionsService;
+
+  private directionsRenderer!: google.maps.DirectionsRenderer;
+
+  private geocoder!: google.maps.Geocoder;
+
+  private routes!: google.maps.DistanceMatrixService;
+
+  public constructor({
+    mapElement,
+    center,
+    zoom,
+    extraLibraries,
+  }: Constructor) {
+    if (map) {
+      this.map = map;
+    }
```

Comment:
Global variable is easiest way, but let's do it in right way. We can use MapService as a singleton or create separate container for map (not global variable) and use composition of Map and MapService. Same for markers